src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -285,6 +285,16 @@ public async Task InitializeAsync()
             _logger.LogTrace($"Done inferring Sql database schema in {timer.ElapsedMilliseconds}ms.");
         }
 
+        /// <inheritdoc />
+        public void InitializeAsync(
+            Dictionary<string, DatabaseObject> entityDatabaseObject,
+            Dictionary<string, string> graphQLStoredProcedureExposedNameToEntityNameMap)
+        {
+            GenerateDatabaseObjectForEntities();
+            EntityToDatabaseObject = entityDatabaseObject ?? EntityToDatabaseObject;
```

Comment:
so it wouldn't be an error if `entityDatabaseObject` was null? Whichever assembly is calling this initialize should be expected to provide non-null params, right?

Reply:
yes technically it shouldnt be null, but this is just check I added if null then it remains unchanged. Currently in SqlMetadtaProvider 

it assigned :         public Dictionary<string, DatabaseObject> EntityToDatabaseObject { get; set; } =
            new(StringComparer.InvariantCulture);
           Hence if incoming is null then we donot assign null but keep it as  empty()